benchmarks/benchmark_prefix_caching.py

```diff
@@ -77,68 +95,102 @@ def sample_requests(
     random.shuffle(dataset)
 
     min_len, max_len = input_length_range
+    assert min_len >= 0 and max_len >= min_len, "input_length_range too small"
 
     # Filter out sequences that are too long or too short
-    filtered_dataset: List[Tuple[str, int, int]] = []
+    filtered_requests: List[Request] = []
+    prefix_token_ids: List[int] = sample_tokens(tokenizer, prefix_len)
+
     for i in range(len(dataset)):
-        if len(filtered_dataset) == num_requests:
+        if len(filtered_requests) == num_requests:
             break
 
         # Tokenize the prompts and completions.
-        prompt = dataset[i][0]
-        prompt_token_ids = tokenizer(prompt).input_ids
+        prompt_token_ids = prefix_token_ids + tokenizer(
+            dataset[i][0]).input_ids
+        prompt = tokenizer.decode(prompt_token_ids)
```

Comment:
It looks a bit weird to prepend a random prefix to a dataset. Should we make `prefix_len` only effective with random dataset?

Reply:
yeah - makes sense to me too.